app/upgrades/v19/upgrades.go

```diff
@@ -53,3 +61,145 @@ func CreateV19UpgradeHandler(
 		return versionMap, err
 	}
 }
+
+// migrateCore1Vesting moves the funds and delegations from the PeriodicVestingAccount -> the new Council (contract address).
+// - Get the Core-1 multisig vesting account
+// - Instantly finish all redelegations, then unbond all tokens.
+// - Send all tokens to the new council (including the previously held balance)
+// - Sum all future vesting periods, then mint and send those tokens to the new council.
+func migrateCore1MultisigVesting(ctx sdk.Context, k *keepers.AppKeepers) {
+	Core1Addr := sdk.MustAccAddressFromBech32(Core1MultisigVestingAccount)
+	CouncilAddr := sdk.MustAccAddressFromBech32(CharterCouncil)
+
+	core1Acc := k.AccountKeeper.GetAccount(ctx, Core1Addr)
+
+	vestingAcc, ok := core1Acc.(*vestingtypes.PeriodicVestingAccount)
+	if !ok {
+		panic(fmt.Errorf("core1Acc.(*vestingtypes.PeriodicVestingAccount): %+v", core1Acc))
+	}
+
+	// SEND TO THE CHARTER
+	prop16Core1Multisig(ctx, k, Core1Addr, CouncilAddr)
+
+	// MINT UNVESTED TOKENS TO THE CHARTER
+	mintUnvestedToCharter(ctx, k, CouncilAddr, vestingAcc)
+
+	// REMOVE VESTING FROM THE CORE1 MULTISIG (set it to the base account, no vesting terms)
+	k.AccountKeeper.SetAccount(ctx, vestingAcc.BaseAccount)
+}
+
+func mintUnvestedToCharter(ctx sdk.Context, k *keepers.AppKeepers, CouncilAddr sdk.AccAddress, vestingAcc *vestingtypes.PeriodicVestingAccount) { // nolint:gocritic
```

Comment:
I'm not sure about this. 

I think all the tokens are already minted, but locked with entries in the vesting module. 

have you checked if in the core-1 multisig the unvested tokens will double? 

i'll try on my local machine too.

Reply:
Ahh this is a good point since locked tokens can still be delegated.

So it is either:
- `InitialGenesisVestingAmount.Sub(currentBal)`
- currentBal, no extra minting

I assume the 2nd is correct here, will dive into deeper

Reply:
yep, probably you need also to delete entries from the vesting module to avoid invariants errors

Reply:
yes, this is done by setting the account to a standard BaseAccount (thus removing all vesting)

Reply:
oh you are right! thanks

Reply:
removed in a01eaf5bd37b113a317f3af5eaf9d4d7fc3f271e as to not double spend the vesting (new amount is roughly 10m JUNO)